### PR TITLE
feat: Gas consumption script

### PR DESCRIPTION
**Pull request by evan-forbes**
## Overview

This PR adds a test suite that we can use to measure the gas consumed by standard txs and modifies the network tests so that we can use custom genesis parameters. We still have to add more tx types and some visualization would be nice.

## Checklist

- [ ] New and updated code has appropriate documentation
- [ ] New and updated code has new and/or updated testing
- [ ] Required CI checks are passing
- [ ] Visual proof for any user facing features like CLI or documentation updates
- [ ] Linked issues closed with keywords

### CODE REVIEW

File: testutil/testnode/full_node.go [hunk 1]
```diff
@@ -135,3 +139,63 @@ func DefaultTendermintConfig() *config.Config {
...
+
+// DefaultNetwork creates an in-process single validator celestia-app network
+// using test friendly defaults. These defaults include fast block times and
+// funded accounts. The returned client.Context has a keyring with all of the
+// funded keys stored in it.
+func DefaultNetwork(t *testing.T, blockTime time.Duration) (cleanup func(), accounts []string, cctx Context) {
+	// we create an arbitrary number of funded accounts
+	for i := 0; i < 300; i++ {
+		accounts = append(accounts, tmrand.Str(9))
+	}
```
Comment (MSevey) [inline testutil/testnode/full_node.go:180]: Append re-allocates memory. So it is more memory performant to initialize the slice and then fill it.
```go
size := 300
accounts = make([]string, 0, size)
for i:=0; i<size; i++ {
  accounts[i] = tmrand.Str(9)
}
```

Comment (evan-forbes) [reply testutil/testnode/full_node.go:180]: thx for the reminder, we actually have a linter for this but I'm not sure why it didn't get hit

**Comment by evan-forbes** [comment]
this is still a draft, and is not worth reviewing atm

**Comment by evan-forbes** [comment]
created a few plots using data collected using the test here
[gas_investigation.pdf](https://github.com/celestiaorg/celestia-app/files/10039380/gas_investigation.pdf)

**Comment by evan-forbes** [comment]
We very likely won't run this test to trace the gas consumption very often, and there is quite a bit of code, so we should just keep them in their own repo.

The only thing this PR should do is enable the investigations to be ran by including the gas consumption monitoring logic imo

**Comment by evan-forbes** [comment]
closing this, as we don't have plans on merging

**Pull request closed without merging**